signal_test.go

```diff
@@ -100,6 +101,10 @@ func TestSignal(t *testing.T) {
 						}
 					}()
 				}
+				var stopCalledTimes atomic.Uint32
```

Comment:
Do we need this to be atomic? From my read, `Stop` shouldn't return until `stopNotify` completes, so I don't think there should be any concurrent access.

Reply:
You're right. Does not need to be atomic.